up-discovery-android/service/src/test/java/org/eclipse/uprotocol/core/udiscovery/NotifierTest.java

```diff
@@ -0,0 +1,173 @@
+    public void setUp() {
+        ShadowLog.stream = System.out;
+        setLogLevel(Log.INFO);
+        mObserverManager = mock(ObserverManager.class);
+        mNotifier = new Notifier(mObserverManager, mUpClient);
+        mObserverMap = buildObserverMap();
+        mUriPath = getUriPath();
+        mAddedUris = getAddedUris();
+    }
+
+    private HashMap<UUri, Set<UUri>> buildObserverMap() {
+        UEntity observerEntity = UEntity.newBuilder().setName(TEST_OBSERVER1).build();
+        UUri observer = UUri.newBuilder().setAuthority(TEST_AUTHORITY).setEntity(observerEntity).build();
+
+        UEntity entity = UEntity.newBuilder().setName(TEST_ENTITY_NAME).build();
+        UUri nodeUri = UUri.newBuilder().setAuthority(TEST_AUTHORITY).setEntity(entity).build();
+
+        Set<UUri> set = new HashSet<>();
+        set.add(observer);
+
+        HashMap<UUri, Set<UUri>> map = new HashMap<>();
+        map.put(nodeUri, set);
+
+        return map;
+    }
+
+    private List<UUri> getUriPath() {
+        UAuthority domainAuthority = UAuthority.newBuilder().setName(TEST_DOMAIN).build();
+
+        UEntity serviceEntity = UEntity.newBuilder().setName(TEST_ENTITY_NAME).build();
+        UEntity versionEntity = UEntity.newBuilder().setName(TEST_ENTITY_NAME).setVersionMajor(1).build();
+
+        UUri domainUri = UUri.newBuilder().setAuthority(domainAuthority).build();
+        UUri deviceUri = UUri.newBuilder().setAuthority(TEST_AUTHORITY).build();
+        UUri entityUri = UUri.newBuilder().setAuthority(TEST_AUTHORITY).setEntity(serviceEntity).build();
+        UUri versionUri = UUri.newBuilder().setAuthority(TEST_AUTHORITY).setEntity(versionEntity).build();
+
+        return List.of(domainUri, deviceUri, entityUri, versionUri);
+    }
+
+    private List<UUri> getAddedUris() {
+        UEntity versionEntity = UEntity.newBuilder().setName(TEST_ENTITY_NAME).setVersionMajor(1).build();
+        UResource resource = UResource.newBuilder().setName(TEST_RESOURCE).setInstance(TEST_INSTANCE_1).build();
+        UUri resourceUri = UUri.newBuilder().setAuthority(TEST_AUTHORITY).setEntity(versionEntity).setResource(resource).build();
+        return List.of(resourceUri);
+    }
+
+    @Test
+    public void testNotifyObserver() {
+        when(mObserverManager.getObserverMap()).thenReturn(mObserverMap);
+        mNotifier.notifyObserversWithParentUri(Notification.Operation.UPDATE, mUriPath);
+        //TODO capture and verify the notification event sent to upClient
+        //when(mUltifiLink.publish(any())).thenReturn(buildStatus(Code.OK));
+    }
+
+    @Test
+    public void testNotifyObserver_debug() {
+        setLogLevel(Log.DEBUG);
+        when(mObserverManager.getObserverMap()).thenReturn(mObserverMap);
+        mNotifier.notifyObserversWithParentUri(Notification.Operation.UPDATE, mUriPath);
+        //TODO capture and verify the notification event sent to upClient
+        //when(mUltifiLink.publish(any())).thenReturn(buildStatus(Code.OK));
```

Comment:
Remov all references to ultifi and why are these all commented out?

Reply:
Done